File: Algorithms/leetcode/Ex_560_Subarray_Sum_Equals_K/Tle_1_Brute_Force.cpp
#include <vector>

using std::vector;

class Tle_1 {
public:
    int subarraySum(vector<int>& nums, int k) {
        int res = 0;
        int len = nums.size();
        for (int start = 0; start < len; start++) {
            for (int end = start; end < len; end++) {
                // 计算 nums[start ... end] 之间的和
                int sum = 0;
                for (int i = start; i <= end; i++) {
                    sum += nums[i];
                }
                if (sum == k) {
                    res += 1;
                }
            }
        }
        return res;
    }
};

File: Algorithms/leetcode/Ex_0454_4Sum_II/Tle_1_Brute_Force.py
class Tle_1:
    def fourSumCount(self, nums1: List[int], nums2: List[int], nums3: List[int], nums4: List[int]) -> int:
        res = 0
        for i in range(len(nums1)):
            for j in range(len(nums2)):
                for k in range(len(nums3)):
                    for l in range(len(nums4)):
                        if sum([nums1[i], nums2[j], nums3[k], nums4[l]]) == 0:
                            res += 1
        return res
